inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -393,43 +393,19 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
     return OK;
 }
 
-StatusCode CNNNetworkNGraphImpl::serialize(const std::string& xmlPath, const std::string& binPath,
-                                           ResponseDesc* resp) const noexcept {
-    auto network = cnnNetwork;
-    if (!network) {
-        // TODO: once Serialization::SerializeV10 supports true IR v10
-        // remove this conversion and WA for execution graph
-        try {
-            bool isExecutionGraph = true;
-            for (const auto & op : _ngraph_function->get_ops()) {
-                auto & rtInfo = op->get_rt_info();
-                if (rtInfo.find(ExecGraphInfoSerialization::PERF_COUNTER) == rtInfo.end()) {
-                    isExecutionGraph = false;
-                    break;
-                }
-            }
-            if (isExecutionGraph) {
-                Serialization::SerializeV10(xmlPath, binPath, (InferenceEngine::ICNNNetwork&)*this);
-                return OK;
-            }
-
-#ifdef ENABLE_V7_SERIALIZE
-            network = std::make_shared<details::CNNNetworkImpl>(*this);
-#endif
-        } catch (const InferenceEngineException& e) {
-            return DescriptionBuffer(GENERAL_ERROR, resp) << e.what();
-        } catch (const std::exception& e) {
-            return DescriptionBuffer(UNEXPECTED, resp) << e.what();
-        } catch (...) {
-            return DescriptionBuffer(UNEXPECTED, resp);
-        }
-    }
-
-#ifdef ENABLE_V7_SERIALIZE
```

Comment:
I propose to remove `ENABLE_V7_SERIALIZE` in separate PR.

At the current moment please use `getFunction()` to get ngraph Function.
And if function is empty please serialize the network to v7 format.

Reply:
Can you explain on which circumstances ngfunction can be empty? This is CNNNetworkNGraphImpl so i thought that by definition it must contain ngfunction.

Reply:
In general it can be empty if conversion to legacy was called. it can be happened if for example user call the `setBatchSize()` method.